bin/ruby-build

```diff
@@ -1256,6 +1256,29 @@ build_package_ldflags_dirs() {
   done
 }
 
+build_package_enable_yjit() {
+  local rustc_installed rustc_ver
+
+  rustc_installed=""
+  if which rustc; then
+    rustc_ver=`rustc --version | cut -c7-10`
```

Comment:
Likely needs a `2>/dev/null || true` or so, like other checks in this file, so there is no extra output if missing.
Could you test what happens if rustc is not installed? One easy way is to change to e.g. `rustcNOTEXIST` here.

Re the `cut -c7-10` there is a builtin bash way to do that, or it could be done by the matching below which seems safer than hardcoded character offsets.

Reply:
I don't think there's an easy built-in bash way to do it, nor a built-in bash comparison for dotted version strings. There's "sort -V" but it's not as broadly available as we'd want. I'll update the matching to do it that way.

Reply:
https://www.baeldung.com/linux/bash-substring#3-using-bashs-substring-expansion is what I meant for the builtin way to substring with offsets.
But yes I think extending the matching to be like `"${rustc_ver}" == "rustc 1."[6789]*` is the way to go.